### PR TITLE
Fix binding observer memory leak

### LINKED ISSUE #7022: fix: memory leak due to design token bindings
# 🐛 Bug Report

Given a component whose template uses a derived token value, as instances of that component are replaced in the DOM, the old, detached instances cannot be garbage collected, i.e. are **leaked**.

## 💻 Repro or Code Sample

See [this codepen](https://codepen.io/m-akinc/pen/JjQzbZj). Each time you click the button, it gets replaced by a new button instance. The old buttons, which are no longer referenced from the code or DOM, are permanently leaked. See below:

fast-memory-leak

## 🤔 Expected Behavior

Detached components should be garbage collected.

## 😯 Current Behavior

Memory is leaked.

## 💁 Possible Solution

The cause seems to be that `DesignTokenNode`s do not tear down their binding observers when their corresponding nodes are detached. I.e. this problem is resolved by adding the below `for` loop to `DesignTokenNode.unbind()`:
```ts
    /**
     * Invoked when the DesignTokenNode.target is detached from the document
     */
    public unbind(): void {
        if (this.parent) {
            const parent = childToParent.get(this)!;
            parent.removeChild(this);
        }
        for (const token of this.bindingObservers.keys()) {
            this.tearDownBindingObserver(token);
        }
    }
```

## 🔦 Context

We have a custom table component with row components that are replaced when scrolled in and out of view. This quickly eats up large amounts of memory.

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 📖 Description

Fixes #7022

### 🎫 Issues

#7022

## 📑 Test Plan

No new tests, since branch is about to be frozen anyway.

### General

- [x] I have included a change request file using `$ npm run change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://github.com/microsoft/fast/blob/master/CODE_OF_CONDUCT.md#our-standards) for this project.

## ⏭ Next Steps

None. Does not apply to mainline.

### CODE REVIEW

**Review by chrisdholt — APPROVED** [review]
Would love to get @janechu's review on this as well. Thanks @m-akinc for this fix!

**Comment by m-akinc** [comment]
> Would love to get @janechu's review on this as well. Thanks @m-akinc for this fix!

She actually approved it a few hours before you. 🙂

**Comment by m-akinc** [comment]
Is this ready to be merged?

**Comment by janechu** [comment]
@m-akinc for some strange reason the checks are not running on this PR, so I took your changes and created a PR myself to see what would happen https://github.com/microsoft/fast/pull/7026 and for whatever reason, we are consistently failing some tests on windows.

**Comment by m-akinc** [comment]
> @m-akinc for some strange reason the checks are not running on this PR, so I took your changes and created a PR myself to see what would happen #7026 and for whatever reason, we are consistently failing some tests on windows.

@janechu Those two slider tests seem suspect. They fail for me locally with slightly different numbers (both with and without my change):
```
Chrome Headless 128.0.0.0 (Windows 10) Slider should constrain and normalize the value when the `step` attribute has been provided and is a float FAILED
        AssertionError: expected 0.4 to equal 0.6
            at dist/esm/__test__/webpack:/dist/esm/slider/slider.spec.js:134:15
            at Generator.next (<anonymous>)
            at fulfilled (dist/esm/__test__/webpack:/node_modules/tslib/tslib.es6.js:71:42)
Chrome Headless 128.0.0.0 (Windows 10) Slider should update the `stepMultiplier` when the `step` attribute has been updated FAILED                                                                                                                         
        AssertionError: expected 4 to equal 6
            at dist/esm/__test__/webpack:/dist/esm/slider/slider.spec.js:142:15
            at Generator.next (<anonymous>)
            at fulfilled (dist/esm/__test__/webpack:/node_modules/tslib/tslib.es6.js:71:42)
```

The tests seem to use pixel values with an assumption that the slider is a certain length in pixels. Maybe that is not reliable. Maybe the default control or browser window width has changed. Given that the fast-element-1 branch is about to be locked down, I'd imagine there's no real drawback to just disabling these two tests if you're convinced they're not failing due to my changes.

**Comment by janechu** [comment]
Since they are flakey I think it's fine if you skip those tests in this PR so they're working locally, see if that also kicks off the build tests and we can just make a note of it. Thanks for your patience!

**Comment by m-akinc** [comment]
@janechu pushed change to skip those two tests. Looks like the build tests need approval from a maintainer to run.

File: packages/web-components/fast-foundation/src/slider/slider.spec.ts
Comment (janechu) [inline packages/web-components/fast-foundation/src/slider/slider.spec.ts:19]:
```suggestion
describe("Slider", () => {
```

**Review by janechu — CHANGES_REQUESTED** [review]
We should remove the only, we're only skipping the 2 tests  I believe.

File: packages/web-components/fast-foundation/src/slider/slider.spec.ts
Comment (m-akinc) [reply packages/web-components/fast-foundation/src/slider/slider.spec.ts]: Thanks! Didn't mean to check that in.

**Review by janechu — APPROVED** [review]

**Comment by m-akinc** [comment]
@janechu It looks like there are still a few open PRs into the `fast-element-1` branch, but can we get another release from that branch in the meantime?

**Pull request merged**
